src/inset_state/inset_state.cpp

```diff
@@ -385,6 +385,33 @@ void InsetState::set_area_errors()
   }
 }
 
+void InsetState::adjust_grid()
+{
+  unsigned int long_graticule_length = std::max(lx_, ly_);
+  double curr_max_area_error = max_area_error().value;
+  unsigned int grid_factor =
+    (long_graticule_length > default_long_graticule_length)
+      ? 2
+      : default_grid_factor;
+  max_area_errors_.push_back(curr_max_area_error);
+  if (
+    n_finished_integrations_ >= 2 &&
+    curr_max_area_error > max_area_errors_[n_finished_integrations_ - 1] &&
+    curr_max_area_error > max_area_errors_[n_finished_integrations_ - 2]) {
+
+    // Multiply grid size with factor
+    std::cout << "Adjusting grid size." << std::endl;
```

Comment:
Can we change std::cout to std::cerr? We want to only use std::cout to output cartogram with -o flag.

Reply:
Alright, will change.